client/ayon_deadline/remote_publish.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pyblish import util
+from ayon_core.lib import Logger
+
+
+def check_results(context, log):
+    error_format = "Failed {{plugin.__name__}}: {{error}}:{{error.traceback}}"
+    for result in context.data["results"]:
+        if result["success"]:
+            continue
+        # Error exit as soon as any error occurs.
+        error_message = error_format.format(**result)
+        log.error(error_message)
+        # 'Fatal Error: ' is because of Deadline
+        raise RuntimeError("Fatal Error: {}".format(error_message))
+
+def remote_publish(log):
+    context = util.collect()
+    for instance in context:
+        if (
+            instance.name == os.environ["AYON_INSTANCE_NAME"] and
```

Comment:
Is this something new? I might need some enlightment... Why we're passing product name and product type instead of passing json with instances that should be processed?

Also why to handle enablid/disabling of the instances out of publishing? They could be disabled/enabled in collect plugin, or? That would help to avoid implementing separate logic for pyblish.

Reply:
I think we should be passing the Created Instances identifiers to the publish script, so that we can do an exact match to a `CreatedInstance` that we want to enable/disable even at the `CreateContext` stage. I'd even disable them then directly in `CreateContext`.

ALSO, It's **absolutely critical** that the "Save Workfile" and "Increment Workfile" logic does not trigger on the farm.
Potentially those should 'skip' in headless mode (easiest) but preferably they explicitly get skipped if the current publish is a 'batch publish' process.

Reply:
>  It's absolutely critical that the "Save Workfile" and "Increment Workfile" logic does not trigger on the farm.

Can be achieved by adding `target = ["local", "farm"]` to them.